### PR TITLE
Add release candidate number to version code

### PR DESCRIPTION
**Pull request by nielsvanvelzen**
See documented code and Kotlin Playground sample which includes a test: https://pl.kotl.in/O_ZNvkGbQ

### CODE REVIEW

File: settings.gradle.kts [hunk 1]
```diff
@@ -1,7 +1,5 @@
 include(":app")
 
-rootProject.name = "Jellyfin"
```
Comment (Maxr1998) [inline settings.gradle.kts:3]: Why remove that? :thinking:

File: buildSrc/src/main/kotlin/VersionUtils.kt [hunk 2]
```diff
@@ -1,29 +1,51 @@
...
         .take(3)
         .toList()
 
-    // Not a valid semver
-    if (parts.size != 3) return null
+    // Parse pre release part (ignore type, only get the number)
+    val buildVersion = versionPreRelease
+        ?.split('.', limit = 2)
+        ?.getOrNull(1)
+        ?.let(String::toIntOrNull)
```
Comment (Maxr1998) [inline buildSrc/src/main/kotlin/VersionUtils.kt:41]: This should have better performance and readability:
```suggestion
    val buildVersion = versionPreRelease?.run {
        substringAfter('.').let(String::toIntOrNull)
    }
```

File: buildSrc/src/main/kotlin/VersionUtils.kt [hunk 3]
```diff
@@ -1,29 +1,51 @@
...
+    val (versionCore, versionPreRelease) = versionName
+        .split("-", limit = 2)
+        .let {
+            when (it.size) {
+                // No pre release part included
+                1 -> arrayOf(it[0], null)
+                // Pre release part included
+                else -> arrayOf(it[0], it[1])
+            }
+        }
```
Comment (Maxr1998) [inline buildSrc/src/main/kotlin/VersionUtils.kt:28]:
```suggestion
    val (versionCore, versionPreRelease) = versionName.run {
        when (val index = indexOf('-')) {
            // No pre-release part included
            -1 -> this to null
            // Pre-release part included
            else -> substring(0, index) to substring(index + 1, length)
        }
    }
```

**Review by Maxr1998 — CHANGES_REQUESTED** [review]

File: settings.gradle.kts (hunk 1 above)
Comment (nielsvanvelzen) [reply settings.gradle.kts:3]: It doesn't do anything except telling IntelliJ/Android Studio to name the project "Jellyfin" in all places (which is annoying when I have multiple Jellyfin projects, just use the directory name please)

**Comment by nielsvanvelzen** [comment]
I did change your suggestions a bit for readability purposes, new sample: https://pl.kotl.in/HHFXfYGOA

File: settings.gradle.kts (hunk 1 above)
Comment (Maxr1998) [reply settings.gradle.kts:3]: Fair enough, we could change that to Jellyfin Android though.

**Review by Maxr1998 — APPROVED** [review]
Yeah, that's also good. You could still pull up the when to the same line, that the closing brackets match the start of the line.

**Comment by nielsvanvelzen** [comment]
The line would be quite long so this looked better to me

**Comment by Maxr1998** [comment]
> The line would be quite long so this looked better to me

It's barely longer than the comment, but still true. Gonna merge then.

**Pull request merged**